### PR TITLE
Redesign of KButton

### PR DESCRIPTION
**Pull request by Pan-Bubr**
### Summary
This implements design changes requested in KHCP-1888.

#### Changes made:
* Add default rounded style of button
* Add medium and large button sizes
* Remove outlinePrimary outlineDanger btnLinkDaner appearances
* Add creation appearance
* Adjust paddings to new redesign

### PR Checklist
- [ ] Does not introduce dependencies
- [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [ ] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [ ] **Framework style:** abides by the essential rules in Vue's style guide
- [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: packages/styles/_variables.scss [hunk 1]
```diff
@@ -29,6 +29,11 @@ $colors: (
   green-400: #19a654,
   green-500: #008036,
 
+  // Creation button - discuss with designer
+  green-creation-1: #008871,
```
Comment (tomaszwylezek) [inline packages/styles/_variables.scss:33]: Why it does not follow same convention as the others? Like naming `green-600` and `green-700`?

File: packages/KButton/KButton.vue [hunk 2]
```diff
@@ -288,6 +297,9 @@ export default {
       @include boxShadow(var(--red-500, color(red-500)), 0, 2px);
     }
   }
+  &.rounded {
+    border-radius: 100px;
```
Comment (tomaszwylezek) [inline packages/KButton/KButton.vue:301]: Maybe better will be use `50%` instead of strict px value?

File: packages/KButton/KButton.vue [hunk 3]
```diff
@@ -237,35 +248,33 @@ export default {
     &:focus {
       @include boxShadow(var(--KButtonDangerBase, var(--red-500, color(red-500))));
     }
-  }
-  &.outline-primary {
-    color: var(--KButtonOutlinePrimaryColor, var(--blue-500, color(blue-500)));
-    border-color: var(--KButtonOutlinePrimaryBorder, rgba(color(blue-500), .4));
-    background-color: var(--white, color(white));
+  }&.creation {
```
Comment (tomaszwylezek) [inline packages/KButton/KButton.vue:251]: Probably it should be in next line

File: packages/KButton/KButton.vue [hunk 4]
```diff
@@ -13,7 +13,7 @@
     :type="type"
     :is="buttonType"
     :to="to"
-    :class="[size === 'default' ? '' : size, {'icon-btn': !hasText && hasIcon}, appearance, caretClasses]"
+    :class="[size === 'default' ? '' : size, {'icon-btn': !hasText && hasIcon}, appearance, caretClasses, rounded ? 'rounded' : '']"
```
Comment (tomaszwylezek) [inline packages/KButton/KButton.vue:16]:
```suggestion
    :class="[size === 'default' ? '' : size, {'icon-btn': !hasText && hasIcon, rounded}, appearance, caretClasses]"
```

File: packages/styles/_variables.scss (hunk 1 above)
Comment (Pan-Bubr) [reply packages/styles/_variables.scss:33]: The discussion about this palette is in progress.

File: packages/KButton/KButton.vue (hunk 4 above)
Comment (Pan-Bubr) [reply packages/KButton/KButton.vue]: The name is addressed

File: packages/KButton/KButton.vue (hunk 3 above)
Comment (Pan-Bubr) [reply packages/KButton/KButton.vue]: Done

File: packages/KButton/KButton.vue (hunk 2 above)
Comment (Pan-Bubr) [reply packages/KButton/KButton.vue:302]: This is a different behaviour. We want the element to have rounded edges and using 50% will make it elliptical.

File: packages/styles/_variables.scss (hunk 1 above)
Comment (Pan-Bubr) [reply packages/styles/_variables.scss]: We are falling back to the existing colors while waiting for the new accessibility-friendly palette

**Comment by kaiarrowood** [comment]
@chrisvdp do you know what Adrian would need in the commit message to change our version to beta when this PR gets merged?

**Comment by chrisvdp** [comment]
> @chrisvdp do you know what Adrian would need in the commit message to change our version to beta when this PR gets merged?

I believe we want to add `preid: 'beta'` to the lerna.json config. This should be it's own PR.

**Comment by kaiarrowood** [comment]
> I believe we want to add `preid: 'beta'` to the lerna.json config. This should be it's own PR.

Adrian, you can ignore this discussion!

**Comment by kaiarrowood** [comment]
Components that look weird now:

image

Do we have a new design for the viewswitcher? May want to pull this one in https://konghq.atlassian.net/browse/KHCP-1906
At the least, for now let's use `small` `outline` `isRounded=false`

image

Let's change all buttons in `KCard` from `secondary` to `outline`

image

Let's change to `small` `outline` `isRounded=false`

**Review by kaiarrowood — APPROVED** [review]
Good to go! great work!

**Pull request merged**